### PR TITLE
Add set.{__iter__,__ior__,__iand__,__isub__,__ixor__}

### PR DESCRIPTION
**Pull request by palaviv**

### CODE REVIEW

File: tests/snippets/set.py [hunk 1]
```diff
@@ -99,17 +99,64 @@ def __hash__(self):
...
 
+a = set([1,2,3])
+a &= set([2,3,4,5])
+assert a == set([2,3])
+try:
+	a &= 1
+except TypeError:
+	pass
+else:
+	assert False, "TypeError not raised"
```
Comment (OddCoincidence) [inline tests/snippets/set.py]: Could you use the `assert_raises` function for these checks? Just makes these a bit less verbose

Comment (palaviv) [reply tests/snippets/set.py]: You cant do assignment inside lambda function so the best I can think about is something like this:
```py
a = [set([1,2,3])]
def f():
    a[0] |= set([4,5])
assert_raises(TypeError, f)
```
I believe the current option is more clear. Any other suggestion?

Comment (OddCoincidence) [reply tests/snippets/set.py]: Oh, I completely missed that it's an assignment! Hopefully in the future we'll have something like `assertRaises` from `unittest`:

```python
with self.assertRaises(TypeError):
    a[0] |= set([4,5])
```

Comment (windelbouwman) [reply tests/snippets/set.py]: The with statement should work, so we could do the same now by implementing a contextmanager.

**Pull request merged**